### PR TITLE
Adds ID card painter

### PR DESCRIPTION
**Pull request by Bawhoppen**
The PDA painter now has the combined functionality to be able to paint ID cards, in the same fashion and painting PDAs.
By default it will only have the basic, prisoner, silver, basic centcom, and captain's colors available but if emagged it will be able to access from a wide range of new ID card sprites.

The nukeops base has also been given an emagged variant incase they want to stealthops.

### CODE REVIEW

**Comment by KazeEspada** [comment]
:-1:

**Comment by TechnoAlchemisto** [comment]
this is why the chameleon kit exists!

**Comment by TechnoAlchemisto** [comment]
Also, the cards aren't painted that color they're made of special space materials to give them the golden/silver gleam

**Comment by Bawhoppen** [comment]
@TechnoAlchemisto no

**Comment by Bawhoppen** [comment]
I didn't know that agent cards had the ability to change sprite, but it doesn't matter anyways considering their more important use is instantly scanning access aswell as ability to change your ID name aswell as job title. This is just a legit way of actually doing it for non-traitors.

**Comment by KazeEspada** [comment]
Except non-traitors can also use the silver cards that spawn on station. Gold and Centcom cards are supposed to be rare as fuck.

**Comment by Bawhoppen** [comment]
So what if they're not rare? There's legitimately no actual reason it should need to be rare.

**Comment by coiax** [comment]
If an assistant breaks into the Hop office and dyes his ID gold, then he deserves the beating he gets from Security.
:+1: nice idea.

**Comment by Anonmare** [comment]
Actually useful for a HoP who needs to promote a new Head of Staff without both:-
A.) Pestering the Cap for his spare silver ID box - You can just give yourself the access but if the Cap's in, he'll no doubt ignore you and start lasering you no matter how much you scream you need his damn box (Source: This has happened to me).
B.) Juggling access levels and crew records. The crew record get's updated to their original ID.

Plus it cuts down on the number of unassigned cards hanging around the office when the Heads inevitably demand demotions to be carried out.

**Comment by xxalpha** [comment]
Let rare things be rare.

:-1:

**Comment by optimumtact** [comment]
>people pretending gold id cards are rare
What

This is a perfectly fine change

**Comment by PKPenguin321** [comment]
yeah, ID cards are only technically rare, but the colors themselves actually do jack shit
:+1: to this, a traitor can now emag the id console thing and have an actually somewhat convincing centcomm officer disguise

**Comment by AdamElTablawy** [comment]
yeah spending 6 tc and breaking into the hop office, risking getting fucked by security and the AI + the HoP to change your ID card isn't really overwriting the existence of the agent ID that lets it change color on the fly, no breakins required

**Comment by Aloraydrel** [comment]
Lore is being ruined!

**Comment by Bawhoppen** [comment]
let this be a reminder that the coders control the lore

**Comment by Bawhoppen** [comment]
Alright, I have fixed some of the problems and now everything works as intended.

Here are the images of the new ID card sprites btw:
ss 2016-04-09 at 10 42 40
I salvaged them from NT-station.

**Comment by PKPenguin321** [comment]
you can disguise your id as an emag :+1:

**Comment by Bawhoppen** [comment]
well no not actually thats just the .dmi file but should I add that to the emagged version's functionality?

**Comment by PKPenguin321** [comment]
id rather have it as default tbh
if you lock it behind emags then it will almost never get used, and if you leave it as a default function then clowns can prank people with fake emags and other shenanigans

File: code/game/machinery/PDApainter.dm [hunk 1]
```diff
@@ -37,15 +40,24 @@
 		//D.name = "PDA Style [colorlist.len+1]" //Gotta set the name, otherwise it all comes up as "PDA"
 		D.name = D.icon_state //PDAs don't have unique names, but using the sprite names works.
 
-		src.colorlist += D
+		src.pdacolorlist += D
+
+
+	for(var/I in typesof(/obj/item/weapon/card/id))
+		var/obj/item/weapon/card/id/D = new I
```
Comment (RemieRichards) [inline code/game/machinery/PDApainter.dm]: OH LORD NO.

Comment (PKPenguin321) [inline code/game/machinery/PDApainter.dm]: que es la problema

**Comment by xxalpha** [comment]
Make an ID card printer that consumes resources instead of a painter.

**Comment by Bawhoppen** [comment]
@xxalpha I can only ask, why?

**Comment by xxalpha** [comment]
Because then you can get rid of the dumb ID boxes and the golden IDs etc will have actual value.

**Comment by Cheridan** [comment]
You also need to use the map merger baw

**Comment by coiax** [comment]
DON'T BE SILLY, 9000 lines changed is TOTALLY NORMAL

**Comment by Bawhoppen** [comment]
I did use the map merger though

**Comment by KorPhaeron** [comment]
Do we really need over colourful IDs on top of everything else? How long until these are made a roundstart prefs

**Comment by Bawhoppen** [comment]
is the slope too slippery kor

**Comment by PKPenguin321** [comment]
He's just having flashbacks to the "IDs start in PDAs" bikeshedding incident

**Comment by duncathan** [comment]
He used the map merger; this is just the initial conversion to TGM.

**Comment by Bawhoppen** [comment]
wait I thought I just fucked up, thanksdunc

**Comment by xxalpha** [comment]
I still think the ability to have different looking IDs should be made costly, to preserve the novelty.

**Comment by Bawhoppen** [comment]
I am dissatisfied with the implementation
I will return to this later perhaps

**Pull request closed without merging**